tests/ui/lint/warn-path-statement.rs

```diff
@@ -14,4 +14,12 @@ fn main() {
 
     let z = (Droppy,);
     z; //~ ERROR path statement drops value
+
+    macro_rules! foo {
+        ($e:expr) => {
+            $e;
```

Comment:
Note: I think you'll need a

```
//~^ ERROR: path statement with no effect
```

annotation here. I guess partly what I had in mind re. macros was also this case, where the suggestion is tied to removing `$e;`, though arguably maybe it's the invocation `foo!(x);` that should be removed.

Again, this is not blocking IMO.